packages/backend/ubuntu-pro_test.go

```diff
@@ -0,0 +1,460 @@
+    }
+    _, err = manager.proServer.DisableInfra(ctx, nil)
+    if err == nil {
+		t.Fatal("I expected an error, but didn't get one")
+    }
+    _, err = manager.proServer.DisableKernelLivePatch(ctx, nil)
+    if err == nil {
+		t.Fatal("I expected an error, but didn't get one")
+    }
+}
+
+func TestEnableService(t *testing.T) {
+    failEnable = false
+    _, err := manager.proServer.EnableEsmApps(ctx, nil)
+    if err != nil {
+		t.Fatal(err)
+    }
+    _, err = manager.proServer.EnableInfra(ctx, nil)
+    if err != nil {
+		t.Fatal(err)
+    }
+    _, err = manager.proServer.EnableKernelLivePatch(ctx, nil)
+    if err != nil {
+		t.Fatal(err)
+    }
+
+    failEnable = true
+    _, err = manager.proServer.EnableEsmApps(ctx, nil)
+    if err == nil {
+		t.Fatal("I expected an error, but didn't get one")
+    }
+    _, err = manager.proServer.EnableInfra(ctx, nil)
+    if err == nil {
+		t.Fatal("I expected an error, but didn't get one")
+    }
+    _, err = manager.proServer.EnableKernelLivePatch(ctx, nil)
+    if err == nil {
+		t.Fatal("I expected an error, but didn't get one")
+    }
+}
+
+func TestIsMachineProAttached(t *testing.T) {
+	getResponse = "true"
+	i, err := manager.proServer.IsMachineProAttached(ctx, nil)
+	if i.GetValue() != true || err != nil {
+		t.Fatal(i.GetValue(), err)
+	}
+
+	getResponse = "false"
+	i, err = manager.proServer.IsMachineProAttached(ctx, nil)
+	if i.GetValue() != false || err != nil {
+		t.Fatal(i.GetValue(), err)
+	}
+
+	getResponse = "error"
+	_, err = manager.proServer.IsMachineProAttached(ctx, nil)
+	if err == nil {
+		t.Fatal("I expected an error, but didn't get one")
+	}
+}
+
+
+func TestIsServiceEnabled(t *testing.T) {
+	getResponse = "disabled"
+	i, err := manager.proServer.IsEsmInfraEnabled(ctx, nil)
+	if i.GetValue() != false || err != nil {
+		t.Fatal(i.GetValue(), err)
+	}
+	i, err = manager.proServer.IsEsmAppsEnabled(ctx, nil)
+	if i.GetValue() != false || err != nil {
+		t.Fatal(i.GetValue(), err)
+	}
+	i, err = manager.proServer.IsKernelLivePatchEnabled(ctx, nil)
+	if i.GetValue() != false || err != nil {
+		t.Fatal(i.GetValue(), err)
+	}
+
+	getResponse = "enabled"
+	i, err = manager.proServer.IsEsmInfraEnabled(ctx, nil)
+	if i.GetValue() != true || err != nil {
+		t.Fatal(i.GetValue(), err)
+	}
+	i, err = manager.proServer.IsEsmAppsEnabled(ctx, nil)
+	if i.GetValue() != true || err != nil {
+		t.Fatal(i.GetValue(), err)
+	}
+	i, err = manager.proServer.IsKernelLivePatchEnabled(ctx, nil)
+	if i.GetValue() != true || err != nil {
+		t.Fatal(i.GetValue(), err)
+	}
+
+	getResponse = "error"
+	i, err = manager.proServer.IsEsmInfraEnabled(ctx, nil)
+	if err == nil {
+		t.Fatal("I expected no error, but didn't get one")
+	}
+	i, err = manager.proServer.IsEsmAppsEnabled(ctx, nil)
+	if err == nil {
+		t.Fatal("I expected no error, but didn't get one")
+	}
+	i, err = manager.proServer.IsKernelLivePatchEnabled(ctx, nil)
+	if err == nil {
+		t.Fatal("I expected no error, but didn't get one")
+	}
```

Comment:
Putting the comment here, but applies to all the if err... checks, I would swap them all for requires as mentioned above :smile: